src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -1666,47 +1757,31 @@ void Mcmcmc::swapRandomChains(void)
         {
             active_chain_index = j;
         }
-        
-        double bj = chain_heats[j];
-        double bk = chain_heats[k];
-        chain_heats[j] = bk;
-        chain_heats[k] = bj;
-        size_t tmp = heat_ranks[j];
-        heat_ranks[j] = heat_ranks[k];
-        heat_ranks[k] = tmp;
-        
-        // also swap the tuning information of each move
-        //            RbVector<Move>& tmp_movesj = chains[j]->getMoves();
-        //            RbVector<Move>& tmp_movesk = chains[k]->getMoves();
-        //            swapMovesTuningInfo(tmp_movesj, tmp_movesk);
-        //            chains[j]->setMoves(tmp_movesj);
-        //            chains[k]->setMoves(tmp_movesk);
-        
-        std::vector<Mcmc::tuningInfo> tmp_mvs_ti = chain_moves_tuningInfo[j];
-        chain_moves_tuningInfo[j] = chain_moves_tuningInfo[k];
-        chain_moves_tuningInfo[k] = tmp_mvs_ti;
-        
-        
+
+        std::swap( chain_heats[j], chain_heats[k] );
+        std::swap( heat_ranks[j], heat_ranks[k] );
+        std::swap( chain_moves_tuningInfo[j], chain_moves_tuningInfo[k] );
```

Comment:
Is this MPI safe? Not sure if I caught the details, but we often runs MC3 with MPI and use exactly as many cores as chains. So was this tested?

Reply:
I constructed an MC3 example and ran it under MPI with 1, 2, and 4 cores.  Given a fixed number of chains, the number of cores does not change the output.  I found and avoided a few mistakes that way.

I don't think any of the current tests use MC3, but we should probably add one that uses it.

Regarding the `std::swap( )` code, it does exact the same thing as the previous code.  It is just shorter.